artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQSession.java

```diff
@@ -392,7 +392,7 @@ public void send(final ProducerInfo producerInfo,
 
       assert clientId.toString().equals(this.connection.getState().getInfo().getClientId()) : "Session cached clientId must be the same of the connection";
       originalCoreMsg.putStringProperty(MessageUtil.CONNECTION_ID_PROPERTY_NAME, clientId);
-
+      originalCoreMsg.putStringProperty(org.apache.activemq.artemis.api.core.Message.PRODUCER_ID, producerInfo.getProducerId().toString());
```

Comment:
you are correct however we already do this with the above connection id, the unused class OpenwireMessage was a wip to fix this. so I think this is the only way to go for now

Reply:
Lets make things better instead of making it worse. Doing the same crap thing 'because someone else did before' only ever leads in the wrong direction.

Field on the message object, e.g the way you did it for the AMQP messages?

Reply:
This is not part of this commit, implementing OpenwireMessage is an orthogonal piece of work which is not part of  this PR

Reply:
Removed as part of previous change, I will talk to @clebertsuconic and send a separate commit to remove OpenWire as not needed